lib/TransferJobs/Transfer.cs

```diff
@@ -216,6 +216,22 @@ public virtual TransferContext Context
             set;
         }
 
+        public IDataMovementLogger Logger
+        {
+            get => _logger ?? NullLogger.Instance;
+            set
+            {
+                if (value == null)
+                {
+                    return;
+                }
+                
+                _logger = value;
```

Comment:
```suggestion
_logger = value ?? NullLogger.Instance;
```
Why? Because someone want to turn off logger and in your code this doesn't happen. Code still uses old logger. In my suggestion you change logger to a `NullLogger`.